lib/src/test/resources/test-cases.json

```diff
@@ -1720,12 +1804,30 @@
       {"tags": "hello"},
       false
     ],
+    [
+      "$all - fail null value",
+      {"tags": {"$all": ["one", "three"]}},
+      {"tags": null},
+      false
+    ],
     [
       "$nor - pass",
       {"$nor": [{"name": "john"}, {"age": {"$lt": 30}}]},
       {"name": "jim", "age": 40},
       true
     ],
+    [
+      "$nor - pass with null value for name",
+      {"$nor": [{"name": "john"}, {"age": {"$lt": 30}}]},
+      {"name": null, "age": 40},
+      true
+    ],
+    [
+      "$nor - pass with null value for age",
+      {"$nor": [{"name": "john"}, {"age": {"$lt": 30}}]},
+      {"name": "jim", "age": null},
+      true
+    ],
```

Comment:
Why it should pass?
If we follow Javascript Style `null` is `0` then `(0 < 30) = true`
`false nor true = not(false or true) = not(true) = false`

Reply:
Hey @Bohdan-Kim, thank for reviewing this PR. Unfortunately I'm ill at the moment. I will check all your comments and update the PR accordingly next week.

Reply:
Hello @nnusse, I hope you doing well. 
Do you mind if we fix failing test by ourself in order to merge the request faster?

Reply:
You are right. I had did a mistake in my implementation that you corrected at the point where you resolved the merge conflict. I will adjust the test definition.

Reply:
I will mark that discussion as resolved. Feel free to give me more feedback if you have any further concerns or questions.